department-of-veterans-affairs/abd-vro
# Enhance Exception Handling for the integration of BIP Claim API in MAS automation

Enhance the BIP exception handling in VRO.

## What was the problem?
Some BIP API call exceptions are not handled, e.g., 404, 401, 500 returns from BIP API service.

Associated tickets or Slack threads:

MCP-2309, MCP-2311, MCP-2310

## How does this fix it?[^1]
Handle the exceptions in GlobalExceptionHandler at controller level, and let unhandled go to Slack channel message.

## How to test this PR
- Step 1. Build the app in local.
- Step 2. Test automatedClaim endpoint with Claim ID: 401, 404, 500. The error messages should be shown in response.


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
